conclave-host/src/main/kotlin/com/r3/conclave/host/internal/attestation/EnclaveQuoteServiceDCAP.kt

```diff
@@ -1,25 +1,26 @@
 package com.r3.conclave.host.internal.attestation
 
 import com.r3.conclave.common.internal.*
-import com.r3.conclave.host.AttestationParameters
 import com.r3.conclave.host.internal.Native
 import com.r3.conclave.host.internal.NativeLoader
 import java.nio.ByteBuffer
 
-class EnclaveQuoteServiceDCAP(val attestationParameters: AttestationParameters.DCAP): EnclaveQuoteService() {
+class EnclaveQuoteServiceDCAP : EnclaveQuoteService() {
+    private val targetInfo = Cursor.allocate(SgxTargetInfo)
 
-    override fun initializeQuote(): Cursor<SgxTargetInfo, ByteBuffer> {
-        val targetInfo = Cursor.allocate(SgxTargetInfo)
-        Native.initQuoteDCAP(NativeLoader.libsPath.toString(), targetInfo.buffer.array())
+    init {
+        Native.initQuoteDCAP(NativeLoader.libsPath.toString(), false, targetInfo.buffer.array())
```

Comment:
The previous code was actually doing it each time we were updating the attestation and luckily we did not have issues :-)
This is meant to be called only once, would you prefer to move it to a companion object anyway?

Reply:
I presume we haven't had issues because we've not tested created two `EnclaveHost` instances in the same process. So yes we should do it, but there's no need to do it now, as you this is how it was before. I also noticed that this class can be an `object` and so we get this singleton loading for free if we just change that.